### PR TITLE
chore(storage-manager): add first-class support for path

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add first class support for `path` as callback
- remove `prefix` prop
- allow `accessLevel` to be `undefined` to support `path` only use cases
- remove default `accessLevel` value

#### Issue #, if available
NA

#### Description of how you validated changes
Manual tests, unit tests
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageManager/types.ts [hunk 1]
```diff
@@ -58,10 +56,10 @@ export interface StorageManagerProps {
    */
   acceptedFileTypes?: string[];
   /**
-   * @deprecated
```
Comment (esauerbo) [inline packages/react-storage/src/components/StorageManager/types.ts:61]: Why are we removing the deprecation warning?

File: packages/react-storage/src/components/StorageManager/StorageManager.tsx [hunk 2]
```diff
@@ -26,63 +27,53 @@ import {
...
-    );
-  }
-
-  if (!prefix && !accessLevel) {
-    // set accessLevel to private to respect the default before adding the `prefix` prop
-    accessLevel = 'private';
-    // accessLevel has been depreacted so only guide the user to pass the `prefix` prop
-    logger.warn('`StorageManager` requires the `prefix` prop to be specified');
+    // eslint-disable-next-line no-console
+    console.warn(MISSING_REQUIRED_PROPS_MESSAGE);
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageManager/StorageManager.tsx:66]:
```suggestion
    logger.warn(MISSING_REQUIRED_PROPS_MESSAGE);
```

**Review by reesscot — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

File: packages/react-storage/src/components/StorageManager/types.ts (hunk 1 above)
Comment (reesscot) [reply packages/react-storage/src/components/StorageManager/types.ts:61]: Let's chat about this offline, there are pros and cons both ways.

File: .changeset/fair-tips-deliver.md [hunk 3]
```diff
@@ -3,4 +3,6 @@
 "@aws-amplify/ui-react": patch
```
Comment (esauerbo) [inline .changeset/fair-tips-deliver.md:3]: would this be considered an additive change? 
```suggestion
"@aws-amplify/ui-react": minor
```

**Review by esauerbo — APPROVED** [review]

File: .changeset/fair-tips-deliver.md (hunk 3 above)
Comment (calebpollman) [reply .changeset/fair-tips-deliver.md:3]: Good catch! Think `patch` is okay cause it's the internal exports but if public `minor` for sure

**Pull request merged**